week-6/Homework/mandatory/1-alarmclock/alarmclock.js

```diff
@@ -1,7 +1,33 @@
 // Implement the missing functionality according to the task description.
 // Hint: Remember which function you need to call when it's time to alarm.
 
-function setAlarm() { }
+
+function setAlarm() {
+  let time = document.getElementById("alarmSet").value
+  let timeMin = Math.floor((time / 60))
+  let timeSecond = (time % 60)
+  alarmInt = setInterval(() => {
+    if (timeSecond === 0 && timeMin > 0) {
+      timeMin--;
+      timeSecond = 59;
+    } else if (timeSecond > 0) {
+      timeSecond--;
+    } else {
+      playAlarm();
+    }
+    document.getElementById("timeRemaining").innerText = `Time Remaining: ${addZero(timeMin)} : ${addZero(timeSecond)}`
+  }, 1000)
+}
+
+let alarmInt;
+window.clearInterval(alarmInt)
+
+function addZero(number) {
+  return (parseInt(number, 10) < 10 ? "0" : "") + number;
+}
+
+
+
```

Comment:
Very good work
Some areas to work on:
1- you need to clear interval in couple of places, creating a function for that can be useful
2- If I set the alarm and then try to set it again, it will cause an issue as interval will be missed up (clearing the interval, in the beginning, will help 
3- stop should pause, meaning I can do click start to continue, else maybe you want stop, to actually clear and stop , or do you want to change start to continue
4- The timer is not start from the same second I set it on, the display show one sec less